### DISCUSSION TITLE
Maximal length of the commit messages

### DISCUSSION THREAD

**Discussion by ismagilli** [magit/magit#5108] [Support] [answered]
Currently, there is a `git-commit-summary-max-length` option in _magit_:

```
User Option: git-commit-summary-max-length
    The intended maximal length of the summary line of commit messages.
    Characters beyond this column are colorized to indicate that this
    preference has been violated.
```

This option limits the length of the first commit line. It would be great to see an option that allows you to set the maximum width for the rest of the commit text (usually the limit for the first line is 50 characters and 72 for the rest).

**Comment by ismagilli** [comment]
I [found](https://emacs.stackexchange.com/a/52478) that I can use `git-commit-finish-query-functions` to check commit message width. But how can I implement this with behaviour similar to `git-commit-summary-max-length`: check lines length and if it necessary highlight in red extra characters?

**Reply by tarsius** [reply]
1. Look at `git-commit-check-style-conventions` to see how such a function should behave if a prefix argument is used and/or a violation is found.
2. Check the buffer for violations, in this case look at each line until you find one that is too long.

**Comment by tarsius** [comment, marked as answer]
> It would be great to see an option that allows you to set the maximum width for the rest of the commit text

That is known as `fill-column` across Emacs.

To display the column specified by that variable, use:

```elisp
(use-package display-fill-column-indicator
  :when (>= emacs-major-version 28)
  :config
  (add-hook 'git-commit-setup-hook
            (lambda ()
              (setq fill-column 72) ; was 70
              (display-fill-column-indicator-mode 1))))
```

That won't highlight the overflowing characters, instead it draws an indicator at column 72.

Use `M-q` (`fill-paragraph`) to perform semi-automatic filling.

**Comment by ismagilli** [comment]
Thanks!

**Reply by tarsius** [reply]
You're welcome!
